### PR TITLE
allow using a toml config file

### LINKED ISSUE #418: Support passing a config file to set flags
In golang Kaspad, you are able to pass in config flags by doing

```
kaspad -C config.conf
kaspad --configfile config.conf
```

Provide that same functionality in rusty kaspad.

Relevant file in rust: `kaspad/src/args.rs`

Requirements:
- file format can be whatever is easiest to support
- cli flags passed explicitly should override the values in the config file

### PR DESCRIPTION
**Pull request by orgr**
resolves #418

### CODE REVIEW

**Review by coderofstuff — CHANGES_REQUESTED** [review]
Thanks for your contribution. In addition to the inline comments, please also check out the TODO below.

## TODO
- [ ] Update the README.md to mention that a config file can now be used, what format it has to be, and show a sample of how to use it.

File: kaspad/src/args.rs [hunk 1]
```diff
@@ -176,6 +178,7 @@ pub fn cli() -> Command {
     let cmd = Command::new("kaspad")
         .about(format!("{} (rusty-kaspa) v{}", env!("CARGO_PKG_DESCRIPTION"), version()))
         .version(env!("CARGO_PKG_VERSION"))
+        .arg(arg!(-C --configfile <CONFIG_FILE> "Path of config file."))
```
[the remark this answers is not in the document]
Comment (orgr) [reply kaspad/src/args.rs:181]: yes, now that I've tested it more thoroughly it appears that defaults take precedence if exist and clap's `get_one()` always return `Some(default)` so `unwrap_or()` didn't work like I'd expect.

I made a kinda hacky fix for it, it's possible that a refactor for the `Args` struct so that it's auto parsed using clap's `Parser` derive macro can be a prettier solution but I didn't want make so many changes.

File: kaspad/src/args.rs [hunk 2]
```diff
@@ -365,22 +368,27 @@ impl Args {
         T: Into<OsString> + Clone,
     {
         let m: clap::ArgMatches = cli().try_get_matches_from(itr)?;
-        let defaults: Args = Default::default();
+        let mut defaults: Args = Default::default();
+
+        if let Some(config_file) = m.get_one::<String>("configfile") {
+            let config_str = fs::read_to_string(config_file)?;
+            defaults = from_str(&config_str).map_err(|_| clap::Error::new(clap::error::ErrorKind::ValueValidation))?;
```
[the remark this answers is not in the document]
Comment (orgr) [reply kaspad/src/args.rs]: the parsing of the config is now stricter and provides better errors, like: 
```
error: failed parsing config file, reason: unknown field `blahblah`, expected one of `appdir`, `logdir`, `no-log-files`, `rpclisten`, `rpclisten-borsh`, `rpclisten-json`, `unsafe-rpc`, `wrpc-verbose`, `log-level`, `async-threads`, `connect-peers`, `add-peers`, `listen`, `user-agent-comments`, `utxoindex`, `reset-db`, `outbound-target`, `inbound-limit`, `rpc-max-clients`, `enable-unsynced-mining`, `enable-mainnet-mining`, `testnet`, `testnet-suffix`, `devnet`, `simnet`, `archival`, `sanity`, `yes`, `externalip`, `perf-metrics`, `perf-metrics-interval-sec`, `block-template-cache-lifetime`, `disable-upnp`, `disable-dns-seeding`, `disable-grpc`, `ram-scale`
```

File: kaspad/src/args.rs [hunk 3]
```diff
@@ -74,7 +76,7 @@ pub struct Args {
 impl Default for Args {
     fn default() -> Self {
         Self {
-            appdir: Some("datadir".into()),
+            appdir: None,
```
Comment (coderofstuff) [inline kaspad/src/args.rs:79]: Why was this change necessary?

**Review by coderofstuff — APPROVED** [review]
Testing again with this new config file:

```
testnet=true
utxoindex=true
disable-upnp=true
perf-metrics=true
unsafe-rpc=true
appdir = "../tn11-test"
testnet-suffix=11
log-level="info,kaspad_lib::daemon=trace"
```

And it's now taking all the params.

File: README.md [hunk 4]
```diff
@@ -248,6 +248,33 @@ The framework is compatible with all major desktop and mobile browsers.
...
+  <summary>
+Use a configuration file
+  </summary>
+
+  ```bash
+cargo run --release --bin kaspad -- --configfile /path/to/configfile.toml
+# or
+cargo run --release --bin kaspad -- -C /path/to/configfile.toml
+  ```
+  The config file should be a list of \<CLI argument\> = \<value\> seperated by newlines.
```
Comment (coderofstuff) [inline README.md:262]: Minor comments:

"seperated by newlines. for example:" -> "sep**a**rated by newlines. **F**or example:"

File: kaspad/src/args.rs (hunk 3 above)
Comment (orgr) [reply kaspad/src/args.rs]: a residual from testing the defaults issue, reverted.

**Review by coderofstuff — APPROVED** [review]
LGTM

**Review by aspect — APPROVED** [review]
